packages/webpack/lib/plugins/assets.js

```diff
@@ -1,34 +1,55 @@
 'use strict';
 
+const { extname } = require('path');
+
 const { RawSource } = require('webpack-sources');
 
 module.exports = class WebpackAssetsPlugin {
-  constructor(options, config, setAssets) {
+  constructor(assets, config, target) {
+    this.assets = assets;
     this.config = config;
-    this.options = options;
-    this.setAssets = setAssets;
+    this.target = target;
   }
   apply(compiler) {
-    compiler.hooks.emit.tap('untool-assets', (compilation) => {
-      const assets = Object.keys(compilation.assets);
-      const chunkNames = Array.from(compilation.namedChunks.keys());
-      const assetsByChunkName = chunkNames.reduce(
-        (result, key) => ({
-          ...result,
-          [key]: compilation.namedChunks.get(key).files,
-        }),
-        {}
+    const { config, target } = this;
+    if (target === 'node') {
+      compiler.hooks.emit.tapPromise('untool-assets', (compilation) =>
+        this.assets.then(
+          (assets) =>
+            (compilation.assets[config.assetFile] = new RawSource(
+              JSON.stringify(assets)
+            ))
+        )
       );
-      if (this.options.static) {
-        this.setAssets(assets, assetsByChunkName);
-      } else {
-        compilation.assets[this.config.assetFile] = new RawSource(
-          JSON.stringify({
-            assets: assets,
-            assetsByChunkName: assetsByChunkName,
-          })
+    } else {
```

Comment:
So, if I understand correctly, this else branch will be executed for `un build` or each watch-run of `un develop` and will then read/reduce the assets from the current compilation and resolve the parent (mixin's) `assets` promise to their result, right?

And that result will then be used in the `if` branch above to write the `assetsFile` to the filesystem (in case of `un build`; or the memory fs, when running `un develop`).

Reply:
You *do* understand correctly 😃